File: disciple/src/main/java/com/monster/disciple/util/NetworkUtil.java
package com.monster.disciple.util;

import android.content.Context;
import android.net.ConnectivityManager;
import android.net.NetworkInfo;

public class NetworkUtil {

    public static boolean isAvailable(Context context) {
        if (context != null) {
            ConnectivityManager connectivityManager = (ConnectivityManager) context
                    .getSystemService(Context.CONNECTIVITY_SERVICE);

            if (null == connectivityManager) return false;

            NetworkInfo mNetworkInfo = connectivityManager.getActiveNetworkInfo();
            if (mNetworkInfo != null) {
                return mNetworkInfo.isAvailable();
            }
        }
        return false;
    }
}

File: disciple/src/main/java/com/monster/disciple/util/NetworkUtil.kt
package com.monster.disciple.util

import android.content.Context
import android.net.ConnectivityManager

fun isAvailable(context: Context?): Boolean {
    if (context != null) {
        val connectivityManager = context
                .getSystemService(Context.CONNECTIVITY_SERVICE) as ConnectivityManager

        val mNetworkInfo = connectivityManager.activeNetworkInfo
        if (mNetworkInfo != null) {
            return mNetworkInfo.isAvailable
        }
    }
    return false
}